examples/multisig/scripts/multisig_goal_sc.js

```diff
@@ -0,0 +1,38 @@
+/**
+ * Description:
+ * This file demonstrates the example to
+   - load a signed lsig from assets and use that lsig to validate transactions
+*/
+const { transferAlgo } = require("./common");
+const { transferMicroAlgos, getMultisigAddress } = require("algob");
+
+async function run(runtimeEnv, deployer) {
+  const masterAccount = deployer.accountsByName.get("master-account")
+  const goldOwnerAccount = deployer.accountsByName.get("gold-owner-account");
+  const johnAccount = deployer.accountsByName.get("john-account");
+  const bobAccount = deployer.accountsByName.get("bob-account");
+
+  /*
+    This example demonstrates a transaction validated by a smart contract signed by a multisig account with threshold of 2
+    User saves the signed logic signature file in assets/ which is extracted and used here.
+  */
+  
+  //Generate multi signature account hash 
+  const addrs =  [goldOwnerAccount.addr, johnAccount.addr, bobAccount.addr]  // you can replace these addresses with your custom addrs for multisig account.
+  const [mparams, kmdMsigAddress] = getMultisigAddress(1, 2, addrs);   // passing (version, threshold, address list)
```

Comment:
why the `mkdMsigAddress` has a `kmd` prefix in name?

Reply:
Since goal uses `kmd`, you can sign a logic signature file. If you try to sign a logic file with an account address not registered using `kmd`, it will throw an error `key does not exist in this wallet`. I faced this issue myself while developing, so added the prefix here. Should i remove it?
Something similar - [](https://forum.algorand.org/t/problem-signing-transaction/180)

Reply:
The individual signatures were created with kmd, but the multisig is universal based on that signatures and logic. If someone else would use the same keys out of kmd then the address will be the same. And this script doesn't uses `KMD`. So I would prefer to remove that prefix and use `msigAddress`